src/components/Calculator.jsx

```diff
@@ -1,77 +1,181 @@
 import React, { Component } from "react";
+import calculate from "../logic/calculate";
 
 class Calculator extends Component {
+  constructor(props) {
+    super(props);
+    this.state = { total: null, next: null, operation: null };
+  }
+
+  componentDidMount = () => {
+    this.setState({
+      total: null,
+      next: null,
+      opreration: null,
+    });
+  };
+
   render() {
+    const updateObject = (e) => {
+      const myObj = calculate(this.state, e.target.textContent);
+      this.setState(myObj);
+    };
+
+    const { next, operation, total } = this.state;
+    const op = operation === "%" ? "mod" : operation;
+    let result = "";
+    if (total) {
+      result = `${total} ${op || ""} ${next || ""}`;
+    } else if (next) {
+      result = `${next} ${op || ""}`;
+    }
+
     return (
       <div className="container">
         <div className="input-field">
-          <p> 765758</p>
+          <p>{result || 0}</p>
         </div>
         <div className="wrapper">
           <div className="row">
-            <button type="button" className="clear-small-button">
+            <button
+              type="button"
+              onClick={updateObject}
+              className="clear-small-button"
+            >
               AC
             </button>
-            <button type="button" className="clear-small-button">
+            <button
+              type="button"
+              onClick={updateObject}
+              className="clear-small-button"
+            >
               +/-
             </button>
-            <button type="button" className="clear-small-button">
+            <button
+              type="button"
+              onClick={updateObject}
+              className="clear-small-button"
+            >
```

Comment:
- [ ] OPTIONAL: This could have been a separate component (`Buttons`) in a separate file and rendered here. Smaller components are easy to fix, reusable, and make the app more scalable. :pray: 😄

Reply:
@GedeonTS, thanks for approving my work! :+1: